README.md

```diff
@@ -1,3 +1,601 @@
+Methods for getting and setting class fields should have prefixes `get` and `set` correspondingly. 
+If a getting method returns a boolean value, it should be prefixed with `is`. 
+For methods naming **lowerCamelCase** notation should be used.
+Method name should fully describe what the method does, be consistent, and unambiguous.
+
+Correct:
+```java
+AuthFilter createAuthenticationFilter();
+```
+
+Incorrect:
+```java
+AuthFilter get_filter();
+```
+
+#### Constants
+Usually, constants are nouns. For naming **CONSTANT_UPPER_CASE** notation should be used - all the letters in upper case,
+each word should be separated by one underscore symbol.
+
+Correct:
+```java
+public static final int NUMBER_OF_ELEMENTS_ON_PAGE = 100;
+```
+
+Incorrect:
+```java
+public static final int numberOfElements = 100;
+```
+
+#### Instance fields, local variables
+Should be in **lowerCamelCase** notation, including *final* fields.
+
+Abbreviations in names should be avoided.
+If it is impossible, for readability purposes **only the first symbol should be in uppercase**.
+
+>For example: **_Http_**_Request_, _Product_**_Dao_**
+
+### JavaDoc
+All public methods and classes should contain JavaDoc. JavaDoc should be explanatory.
+
+#### Formatting
+It is a good practice to separate paragraphs with `<p>`. Sentences should end with the dot symbol.
+
+One-sentenced JavaDoc is allowed if it is a class JavaDoc or there are no parameters in a method.
+
+Correct:
+```java
+/** Animal implementation. */
+public class Cat implements Animal { }
+```
+
+Incorrect:
+```java
+/**
+ * Animal implementation.
+ * <p/>
+ */
+public class Cat implements Animal { }
+```
+
+#### Tags
+Any tags used in JavaDoc should not be left empty.
+
+All tags should follow the order:
+1. **@param** description of a method parameter
+2. **@return** description of a return type
+3. **@throws** exception class name, description of reasons, leading to this exception
+4. **@see** reference to a class, method, field, constant, etc.
+5. **@since** version (from which version tagged element started to be available)
+6. **@deprecated** additional details, including what should be used instead
+
+* To set links to class, field, method, or constant `{@link SomeClass}` tag can also be used.
+* To show text, which should not be interpreted as HTML, the tag `{@code text}` can be used.
+
+>Documentation on [Tag Conventions](https://www.oracle.com/technetwork/articles/java/index-137868.html).
+
+#### Example
+
+```java
+/**
+ * Creates a concrete book file reader by the specified file path. A concrete implementation is chosen based
+ * on the file's extension.
+ *
+ * @param bookFile file referencing a book
+ * @return BookFileReader instance
+ * @throws IllegalArgumentException      if the specified file does not exist
+ * @throws UnsupportedOperationException if an extension of the file is not supported
+ * @see TxtFileReader
+ */
+public static BookFileReader createInstance(final File bookFile) {
```

Comment:
Please, don't use the `final` keyword in a method signature.

Reply:
done